tracer/src/Datadog.Trace/Iast/Propagation/PropagationModuleImpl.cs

```diff
@@ -44,6 +45,104 @@ public static void AddTainted(string? input, Source source)
         }
     }
 
+    public static object? PropagateResultWhenInputTainted(string result, object? firstInput, params object[]? otherInputs)
```

Comment:
would it be possible to use generics here to avoid boxing, especially when instrumenting strings so much 🤔

Reply:
I was thinking about it during this whole PR... This was a compromise of readability vs performance. I could, as in other methods, write a method with more parameters to avoid boxing, like we do in string.concat methods... I will update the code

Reply:
I have updated the code